### DISCUSSION TITLE
Remove the Entry from ChildTable

### DISCUSSION THREAD

**Discussion by Pritesh-1011** [openthread/openthread#8139] [Q&A] [answered]
Hello Openthread Community,

I want to remove the entry of the child node from the child table of the leader before commissioning the same child into the openthread network.

For ex. Currently, I have the child table as below on the leader.

> child table
| ID  | RLOC16 | Timeout    | Age        | LQ In | C_VN |R|D|N|Ver|CSL|QMsgCnt| Extended MAC     |
+-----+--------+------------+------------+-------+------+-+-+-+---+---+-------+------------------+
|   1 | 0xc801 |        240 |         24 |     3 |  131 |1|0|0|  3| 0 |     0 | 4ecede68435358ac |
Done

Now I have done the factory reset of the same child node and again start the commissioning. Now this older entry will last here till its Timeout but I want to remove this entry of the child node before commissioning it again.

Can I do this? If yes then how can I remove this explicitly using any API?

**Comment by jwhui** [comment, marked as answer]
Disabling and enabling the Thread interface using the [`otThreadSetEnabled()`](https://openthread.io/reference/group/api-thread-general#otthreadsetenabled) API should clear the child table.

**Reply by Pritesh-1011** [reply]
Thank you for your quick reply.

But If I have multiple devices in the child table then by disabling and enabling the thread interface, their entry will be also removed. So I want to remove the particular device entry from the table.

**Reply by jwhui** [reply]
Can you provide more detail about your use case? Why do you want to remove a specific child node and recommission? If this is just for demo purposes, one option is to set the Child Timeout to a smaller value so that the child entry times out more quickly on the parent.

**Reply by Pritesh-1011** [reply]
Okay. So in my current system, I have many numbers of sleepy node devices which are connected to the gateway and the wakeup time of the sleepy devices is 1 hour. Now if I set the child Timeout to a smaller value for ex. 60 sec, then when the device went to sleep mode then we are receiving the child removed callback on the gateway after 60 sec, and when the device wakeup we are receiving the node connected callback. This leads to wrong information about the child node's offline/online status. So we increased the child Timeout to 1 hour and with this we are only receiving the child removed callback if the device won't wake up after 1 hour. This solved the offline/online issue.

But now with these changes, We have another issue. If the device child Timeout is set to 1 hour and during this 1 hour if we factory reset the node and again try to commission then we are not able to connect the node with the gateway till its old entry is removed from the child table. So that's why we need to remove its entry from the child table first, then we can proceed with the reprovision.

**Reply by abtink** [reply]
> ... if we factory reset the node and again try to commission then we are not able to connect the node with the gateway till its old entry is removed from the child table. So that's why we need to remove its entry from the child table first, then we can proceed with the reprovision.

If you factory reset the node, it should get a new (random) MAC address and show up as if it is a new/different device. I expect this to work independent of state of child table. Do you maybe persist and/or reuse some info during/after factory reset?

**Reply by Pritesh-1011** [reply]
Yes, I have modified the MAC address to get some unique Identification in the child table using the below code while commissioning.

**otExtAddress extAddress;
otLinkGetFactoryAssignedIeeeEui64(instance, &extAddress);
otLinkSetExtendedAddress(instance, &extAddress);**

**Reply by jwhui** [reply]
Fixing the IEEE 802.15.4 Extended Address even across factory resets is not conformant to the Thread Specification. Allowing OpenThread to generate a new Extended Address after factory reset will resolve your child table issue.

If you need a unique identifier, I suggest communicating one at the application layer. You can use the IEEE EUI-64 as your application layer unique identifier.
